src/components/molecules/ScheduleEvent/ScheduleEvent.tsx

```diff
@@ -0,0 +1,108 @@
+import React, {
+  MouseEventHandler,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import classNames from "classnames";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBookmark as solidBookmark } from "@fortawesome/free-solid-svg-icons";
+import { faBookmark as regularBookmark } from "@fortawesome/free-regular-svg-icons";
+
+import { PersonalizedVenueEvent } from "types/venues";
+
+import { WithVenueId } from "utils/id";
+import { isTruthy } from "utils/types";
+import { isEventLive } from "utils/event";
+
+import { ONE_HOUR_IN_MINUTES } from "utils/time";
+
+import { saveEventToProfile } from "api/profile";
+import { useUser } from "hooks/useUser";
+
+import { calcStartPosition, HOUR_WIDTH } from "../Schedule/Schedule.utils";
+
+import "./ScheduleEvent.scss";
+
+export interface ScheduleEventProps {
+  event: WithVenueId<PersonalizedVenueEvent>;
+  scheduleStartHour: number;
+  isUsers?: boolean;
+}
+
+export const ScheduleEvent: React.FC<ScheduleEventProps> = ({
+  event,
+  scheduleStartHour,
+  isUsers,
+}) => {
+  const [isBookmarked, setBookmark] = useState(event.isSaved);
+
+  useEffect(() => {
+    setBookmark(event.isSaved);
+  }, [event.isSaved]);
+
+  const { userId } = useUser();
+
+  const containerClasses = useMemo(
+    () =>
+      classNames(
+        "ScheduleEvent",
+        {
+          "ScheduleEvent--live": isEventLive(event),
+        },
+        { "ScheduleEvent--users": isTruthy(isUsers) }
+      ),
+    [event, isUsers]
+  );
+
+  const eventWidth = useMemo(
+    () => (event.duration_minutes * HOUR_WIDTH) / ONE_HOUR_IN_MINUTES,
+    [event]
+  );
+  const bookmarkEvent = useCallback(() => {
+    event.isSaved = !event.isSaved;
+
+    if (userId && event.id) {
+      saveEventToProfile({
+        venueId: event.venueId,
+        userId: userId,
+        removeMode: !event.isSaved,
+        eventId: event.id,
+      });
```

Comment:
you can just pass the event? Also, a suggestion:
rename the current function into `updatePersonalisedSchedule` and reuse it in the two fn named: addEvent, removeEvent and they will receive two props: event, userId

Reply:
renamed function: https://github.com/sparkletown/sparkle/pull/1302/commits/eedc4abb3203238d964e4b7da4b8d18a0f3ceb29
but haven't splitted the functionality into addEvent, removeEvent yet.